src/wizard/internal.tsx

```diff
@@ -183,3 +185,43 @@ export default function InternalWizard({
     </div>
   );
 }
+
+/**
+ * This hook observes the step configuration and events the `funnelChange` event when the steps change.
+ */
+function useFunnelChangeEvent(funnelInteractionId: string | undefined, steps: WizardProps['steps']) {
+  const listenForStepChanges = useRef(false);
+
+  useEffect(() => {
+    // We prevent emitting the event before the funnel has stabilised.
+    const handle = setTimeout(() => (listenForStepChanges.current = true), 50);
```

Comment:
What does "stabilised" mean? And how certain are we that 50 is the right magic number here?

Reply:
We want to give some time for updates that happen directly after rendering, for example if some part of the step configuration is loaded asynchronously by the customer or for our effects to detect nested funnels after the first render — changes that are technically not part of the initial appearance, but from the user perspective happen invisibly because they're so fast. The number 50 is somewhat arbitrary, it is chosen so that it is shorter than any reasonable human interaction that we would legitimately count as a change to the configuration, and long enough that React can render at least once and e.g. run any asynchronous effects.